collect_app/src/main/java/org/odk/collect/android/tasks/SaveToDiskTask.java

```diff
@@ -258,30 +259,44 @@ protected void extractGeometryValues(FormInstance instance, ContentValues values
             XPathExpression xpath = XPathParseTool.parseXPath(geometryXpath);
             EvaluationContext context = new EvaluationContext(instance);
             Object result = xpath.eval(instance, context);
-            Timber.w("XPath %s -> %s", geometryXpath, result);
             if (result instanceof XPathNodeset) {
                 XPathNodeset nodes = (XPathNodeset) result;
                 // For now, only use the first node found.
                 TreeElement element = instance.resolveReference(nodes.getRefAt(0));
                 IAnswerData value = element.getValue();
-                Timber.w("value = %s", value);
                 if (value instanceof GeoPointData) {
-                    JSONObject json = toGeoJson((GeoPointData) value);
                     try {
+                        JSONObject json = toGeoJson((GeoPointData) value);
                         String type = json.getString("type");
-                        values.put("geometry", json.toString());
-                        values.put("geometryType", type);
-                    } catch (JSONException e) { }
-                    return;
+                        values.put(InstanceColumns.GEOMETRY, json.toString());
+                        values.put(InstanceColumns.GEOMETRY_TYPE, type);
+                        Timber.i("Geometry for \"%s\" instance found at %s: %s",
+                            instance.getName(), geometryXpath, json);
+                    } catch (JSONException e) {
+                        Timber.w("Could not convert GeoPointData %s to GeoJSON", value);
+                    }
                 }
             }
         } catch (XPathException | XPathSyntaxException e) {
             Timber.w(e, "Could not evaluate geometry XPath %s in instance", geometryXpath);
         }
     }
 
-    protected JSONObject toGeoJson(GeoPointData data) {
-        return new JSONObject();
+    protected JSONObject toGeoJson(GeoPointData data) throws JSONException {
+        // The four (nameless, undocumented) fields of a GeoPointData record are
```

Comment:
Maybe point to https://opendatakit.github.io/xforms-spec/#data-types? Ok, ok, I'll admit it was only added some months ago after your work on other geo so "undocumented" used to be accurate. 🤦🏻‍♀️

Reply:
Sorry about being snarky here—it wasn't helpful. 😓

Reply:
Haha, no worries, it was earned. 😄